lib/easypost/shipment.rb

```diff
@@ -6,6 +6,13 @@ def get_rates(params={})
     return self
   end
 
+  def get_smartrates
+    response = EasyPost.make_request(:get, url + '/smartrate', @api_key)
+    self.refresh_from(response, @api_key, true)
+
+    return self
```

Comment:
uh, I don't think you want to do it this way. You really want to set an attribute on the shipment object called `result` containing the smartrate properties?

Reply:
I'm not sure I follow, is this not the same pattern as the Python CL?

Reply:
Do you mean we should instead be returning the direct values of `result` so that they can instead be accessed like `smartrates[0]...` instead of `smartrates['result'][0]`? If so, I'm in favor, but we'd want to also correct it in the Python CL.

Reply:
So, `self.refresh_from` updates the Shipment object with the dict that you pass it. Most of our API methods return a full Shipment, so that's okay. However, in the case of the smartrate request, it doesn't reutrn the full shipment; it returns just the smartrate parts in a dict with the key `result`.

I think either you should be tweaking the dict so it makes sense and doing something like `self.refresh_from({'smartrate' => response['result']}, ...)` or just not calling `self.refresh_from` at all.

Reply:
Ah, thank you for the clarification! I've removed the `refresh_from` completely.

Reply:
I might still recommend stripping off the `['result']` to make it more ergonomic?

Reply:
@Roguelazer agree; although, I'll check with product to determine the best approach.

Reply:
After talking with Product, they want to leave the result key so that the CLs response looks the same as the bare API, especially now that customers are using this functionality already, it's been announced and docs have the result key, and the Python library keeps the result key. In a perfect world, the key probably wouldn't be there at all and it'd just return a list of objects.

Reply:
That is not something you should talk to product about.

It's a bad API design as it stands. It's completely an engineering decision. I think there's no reason not to make the API better here.